newm-server/src/main/kotlin/io/newm/server/aws/MdcSetup.kt.kt

```diff
@@ -0,0 +1,32 @@
+package io.newm.server.aws
+
+import org.slf4j.MDC
+import aws.sdk.kotlin.services.ec2.Ec2Client
+import aws.sdk.kotlin.services.ec2.model.DescribeInstancesRequest
+import org.koin.core.qualifier.named
+
+suspend fun setInstanceId() {
+    val instanceId = getInstanceIdFromAws() // Replace with your method to get the instance ID
+    MDC.put("instanceId", instanceId)
+}
+
+suspend fun getInstanceIdFromAws(): String? {
+    var instanceId: String? = null
+    val awsRegion = named("aws.region")
+
+    val request =
+        DescribeInstancesRequest {
+            maxResults = 6
+        }
+
+    Ec2Client { region = awsRegion.toString() }.use { ec2 ->
+        val response = ec2.describeInstances(request)
+        response.reservations?.forEach { reservation ->
+            reservation.instances?.forEach { instance ->
```

Comment:
Not sure about how many instances we have in newm? Is there another approach to find these instances?

Reply:
This looks like it gives us all the instances in our account. We need to find the actual instance we're running on. I'm not sure how to do that.

Reply:
@saif-software-developer Check with @jayvaznewm what is the role name of the instances we want, then you can do something like:
```
        val request = DescribeInstancesRequest
            .builder()
            .filters(
                Filter
                    .builder()
                    .name("tag:Role")
                    .values(role)
                    .build()
            ).build()

        val instanceIds = ec2Client
            .describeInstances(request)
            .reservations()
            .flatMap { it.instances() }
            .map { it.instanceId() }

```

Reply:
One of the following tags should work:

```
Name=GarageNewmServer-env-jre21
elasticbeanstalk:environment-name=GarageNewmServer-env-jre21
```

Please note - that those tags are generated based on the Elasticbeanstalk environment name. If you'd like we could add an explicit tag for this purpose as well ( the only benefit would be to protect against Elasticbeanstalk's auto tagging pattern - which is unlikely)